clang/test/AST/dfp.c

```diff
@@ -0,0 +1,40 @@
+// RUN: %clang_cc1 -x c -ast-dump -fexperimental-decimal-floating-point %s | FileCheck %s --strict-whitespace
+
+_Decimal32 x;
+_Decimal64 y;
+_Decimal128 z;
+
+//CHECK:      |-VarDecl {{.*}} x '_Decimal32'
+//CHECK-NEXT: |-VarDecl {{.*}} y '_Decimal64'
+//CHECK-NEXT: |-VarDecl {{.*}} z '_Decimal128'
+
+void f(void) {
+  _Decimal32 x = 0.0DF;
+  _Decimal32 x2 = 1234.0DF;
+  _Decimal32 x3 = 945.75DF;
+  _Decimal32 x4 = 1E-6DF; // DEC32_EPSILON
+  _Decimal32 x5 = 9.999999E40DF;
+  _Decimal32 x6 = 100000.0DF;
+  _Decimal32 x7 = 9.999999E96DF; // DEC32_MAX
+  _Decimal32 x8 = 1E-95DF; // DEC32_MIN
+  _Decimal32 x9 = 0.000001E-95DF; // DEC32_TRUE_MIN
+}
+
+//CHECK:      | `-VarDecl {{.*}} x '_Decimal32'
+//CHECK-NEXT: |   `-FloatingLiteral {{.*}} '_Decimal32' 0E+0 
```

Comment:
If I'm not mistaken, this value is supposed to be 0E-1, not 0E0 (as the final digit is the tenths digit, so the preferred exponent should be -1, not 0).

Reply:
Yeah, I checked out the values using `bid32_from_string` and `bid32_to_string` manually from a stand alone C++ program and `x` and `x9` do not match but the rest do.

Let me see if I can figure out what is going with these two.